src/resources/2013/Florist.ts

```diff
@@ -0,0 +1,364 @@
+import {
+  floristAvailable,
+  getFloristPlants,
+  myLocation,
+  visitUrl,
+} from "kolmafia";
+import { get } from "../../property";
+
+type environment = "outdoor" | "indoor" | "underground" | "underwater";
+
+class Flower {
+  name: string;
+  id: number;
+  environment: environment;
+  modifier: string;
+  territorial: boolean;
+
+  constructor(
+    name: string,
+    id: number,
+    environment: environment,
+    modifier: string,
+    territorial = false
+  ) {
+    this.name = name;
+    this.id = id;
+    this.environment = environment;
+    this.modifier = modifier;
+    this.territorial = territorial;
+  }
+
+  isPlantedHere(location = myLocation()): boolean {
+    return getFloristPlants()[location.toString()].includes(this.name);
+  }
+
+  available(location = myLocation()): boolean {
+    return (
+      this.environment === (location.environment as environment) &&
+      !get("_floristPlantsUsed").includes(this.name) &&
+      !this.isPlantedHere(location)
+    );
+  }
+
+  dig(): boolean {
+    if (!this.isPlantedHere()) return false;
+    const flowers = getFloristPlants()[myLocation().toString()];
+    if (!flowers[2]) return false;
+    const plantNumber = getFloristPlants()[myLocation().toString()].indexOf(
```

Comment:
All of this `getFloristPlants()[myLocation().toString()].whatever` probably needs to be extracted out into a shared function(s) to handle undefineds

Reply:
This one specifically can stay, but the others are extracted